build/webpack.js

```diff
@@ -260,6 +261,10 @@ export default async function getBaseWebpackConfig (dir: string, {dev = false, i
       new webpack.DefinePlugin({
         'process.env.NODE_ENV': JSON.stringify(dev ? 'development' : 'production')
       }),
+      // This is used in client/dev-error-overlay/hot-dev-client.js to replace the dist directory
+      !isServer && dev && new webpack.DefinePlugin({
```

Comment:
Let’s use the environment plugin for this: https://webpack.js.org/plugins/environment-plugin/

(New in Webpack 4)

Reply:
We're not passing `process.env`, we're passing a string.

Reply:
And it abstracts DefinePlugin, so I'd rather use the lower level plugin.